knowledge_repo/app/templates/index-base.html

```diff
@@ -11,13 +11,46 @@
   {% endfor %}
 {% endmacro %}
 
+{% macro pagination(max_pages=10, extremes=True, autohide=True) %}
+  {% if feed_params %}
+    {% set start = feed_params['start'] | default(0) %}
+    {% set results = feed_params['results'] | default(10) %}
+    {% set page = 1 if (start == 0) else (start//results + 1) %}
+    {% set page_count = feed_params['page_count'] | default(1) %}
+
+    {% if autohide and page_count > 1 %}
+      {% set page_nums = pagination_pages(current_page=page, page_count=page_count, max_pages=max_pages, extremes=extremes) %}
+
+      <div class='pagination-bar' role="group">
+        <a href="{{ "javascript:;" if (page == 1) else modify_query(start=(page-2)*results) }}" class="pagination-stepper {{ "disabled" if page == 1 }}">
```

Comment:
I do not agree.

> The href is a javascript void function rather than # because that way it does nothing rather than sending people to the top of the page, which was the previous behaviour (and annoying).

Correct. That being said, you could easily do something like `<a href=# onClick='return false'>`, which would prevent the default behavior. 

In my opinion, if you wanted to do this in Python, then you can precompute the links server-side, and send a dictionary through, with the jinja template doing something like 
`<a href={{ page_url[page]}}` and then have the optional  `onClick` attribute depending on whether it was the first page or not. 

Obviously, however, pre-computing it is non-ideal, which is why I believe you should actually do it in Javascript, on pageload (similar to how it was done). You would then avoid the `onClick` attribute issue.